qt/aqt/editor.py

```diff
@@ -1585,28 +1586,34 @@ def _processImage(self, mime: QMimeData, extended: bool = False) -> str | None:
 
     def contextMenuEvent(self, evt: QContextMenuEvent) -> None:
         m = QMenu(self)
-        self._maybe_add_cut_action(m)
-        self._maybe_add_copy_action(m)
+        if self.hasSelection():
+            self._add_cut_action(m)
+            self._add_copy_action(m)
         a = m.addAction(tr.editing_paste())
         qconnect(a.triggered, self.onPaste)
-        self._maybe_add_copy_image_action(m)
+        if self._opened_context_menu_on_image():
+            self._add_image_menu(AnkiWebView(self), m)
         gui_hooks.editor_will_show_context_menu(self, m)
         m.popup(QCursor.pos())
 
-    def _maybe_add_cut_action(self, menu: QMenu) -> None:
-        if self.hasSelection():
-            a = menu.addAction(tr.editing_cut())
-            qconnect(a.triggered, self.onCut)
+    def _add_cut_action(self, menu: QMenu) -> None:
+        a = menu.addAction(tr.editing_cut())
+        qconnect(a.triggered, self.onCut)
 
-    def _maybe_add_copy_action(self, menu: QMenu) -> None:
-        if self.hasSelection():
-            a = menu.addAction(tr.actions_copy())
-            qconnect(a.triggered, self.onCopy)
+    def _add_copy_action(self, menu: QMenu) -> None:
+        a = menu.addAction(tr.actions_copy())
+        qconnect(a.triggered, self.onCopy)
 
-    def _maybe_add_copy_image_action(self, menu: QMenu) -> None:
-        if self._opened_context_menu_on_image():
-            a = menu.addAction(tr.editing_copy_image())
-            qconnect(a.triggered, self.on_copy_image)
+    def _add_image_menu(self, webview: AnkiWebView, menu: QMenu) -> None:
+        a = menu.addAction(tr.editing_copy_image())
+        qconnect(a.triggered, self.on_copy_image)
+
+        if is_win or is_mac:
```

Comment:
Any feature we add should work across all OSes - is there a reason why non-Mac/Windows has been excluded here?

Reply:
It would definitely be better to make a feature work on all platforms. But, I have no idea how to make it work with all the different file managers in Linux. Even if this doesn't work on Linux, it is still an improvement especially when you consider that Windows + Mac users comprise most of the Anki's userbase.

Reply:
"We can't do it on Linux easily" is a reasonable answer for now :-) Thanks for clarifying.